### PR TITLE
New parametrisation for HyperbolicSpace

### PR DESCRIPTION
**Pull request by hz789**

### CODE REVIEW

File: geomstats/geometry/hyperbolic_space.py [hunk 1]
```diff
@@ -321,42 +357,60 @@ def log(self, point, base_point):
...
-                  + INV_TANH_TAYLOR_COEFFS[5] * angle ** 6
-                  + INV_TANH_TAYLOR_COEFFS[7] * angle ** 8)
+            log = (gs.einsum('ni,nj->nj', coef_1, point) -
+                   gs.einsum('ni,nj->nj', coef_2, base_point))
+            return log
 
-        # This avoids dividing by 0.
-        angle += mask_0_float * 1.
+        if self.point_type == 'poincare':
+            kpx = self.add(-base_point, point)
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/geometry/hyperbolic_space.py]: what is add?

[the remark this answers is not in the document]
Comment (hz789) [reply geomstats/geometry/hyperbolic_space.py]: > what is add?

Add function was added as a seperate function in HyperbolicMetric

File: geomstats/geometry/hyperbolic_space.py [hunk 2]
```diff
@@ -265,45 +276,70 @@ def exp(self, tangent_vec, base_point):
...
-        coef_2 += mask_else_float * (
-            (gs.sinh(norm_tangent_vec) / (norm_tangent_vec)))
-
-        exp = (gs.einsum('ni,nj->nj', coef_1, base_point)
-               + gs.einsum('ni,nj->nj', coef_2, tangent_vec))
-
-        hyperbolic_space = HyperbolicSpace(dimension=self.dimension)
-        exp = hyperbolic_space.regularize(exp)
-        return exp
+            if (0 != len((norm_tangent_vector == 0).nonzero())):
```
[the remark this answers is not in the document]
Comment (hz789) [reply geomstats/geometry/hyperbolic_space.py]: Must test condition still

File: geomstats/geometry/hyperbolic_space.py [hunk 3]
```diff
@@ -321,42 +358,90 @@ def log(self, point, base_point):
...
-               - gs.einsum('ni,nj->nj', coef_2, base_point))
-        return log
+            norm_base_point = base_point.norm(2,
+                                              -1, keepdim=True).expand_as(
+                add_base_point)
+
+            res = (1 - norm_base_point ** 2) * \
+                  ((gs.arc_tanh(norm_add))) * (add_base_point / norm_add)
+
+            res[norm_add == 0] = 0
```
Comment (ninamiolane) [inline geomstats/geometry/hyperbolic_space.py]: use gs.isclose(, 0) instead of == 0

File: geomstats/geometry/hyperbolic_space.py [hunk 4]
```diff
@@ -321,42 +358,90 @@ def log(self, point, base_point):
...
+        mobius_add : array-like, shape=[n_samples, 1]
+                           or shape=[1, 1]
+        """
+        norm_point_a = gs.sum(point_a ** 2, dim=-1,
+                              keepdim=True).expand_as(point_a)
+        norm_point_b = gs.sum(point_b ** 2, dim=-1,
+                              keepdim=True).expand_as(point_a)
+        sum_prod_a_b = (point_a * point_b).sum(-1,
+                                               keepdim=True).expand_as(point_a)
+        return ((1 + 2 * sum_prod_a_b + norm_point_b) * point_a +
```
Comment (ninamiolane) [inline geomstats/geometry/hyperbolic_space.py]: This expression is too complicated: break it down in several lines for readability

**Pull request merged**